src/services/dynamo-database-service.ts

```diff
@@ -100,15 +100,29 @@ export class DynamoDatabaseService {
         ':ttl': { N: ttl.toString() },
         ':false': { BOOL: false },
       },
-      ConditionExpression: 'attribute_not_exists(isAccountDeleted) OR isAccountDeleted = :false',
+      ConditionExpression:
+        'attribute_exists(pk) AND (attribute_not_exists(isAccountDeleted) OR isAccountDeleted = :false)',
     };
     const command = new UpdateItemCommand(commandInput);
-    const response = await this.dynamoClient.send(command);
-    if (!response) {
-      const errorMessage = 'DynamoDB may have failed to update items, returned a null response.';
-      logger.error(errorMessage);
-      logAndPublishMetric(MetricNames.DB_UPDATE_ERROR);
+    try {
+      const response = await this.dynamoClient.send(command);
+      logger.info(`${LOGS_PREFIX_SENSITIVE_INFO} Account ${userId} marked as deleted.`);
+      if (!response) {
+        const errorMessage = 'DynamoDB may have failed to update items, returned a null response.';
+        logger.error(errorMessage);
+        logAndPublishMetric(MetricNames.DB_UPDATE_ERROR);
+        return;
```

Comment:
should throw an error here as well as the db responded with nothing so we are unsure if the update occurred, therefore redrive.
```suggestion
        throw new Error('DB returned a null response when attempting an Update.');
```

Reply:
In the UT response can't be null or undefined, so it's very unlikely to go into the block. I remember Chiara mentioning maybe removing that !response block? Otherwise happy to change it, but I don't seem to be able to test that error in the UT- it won't let me assert a null and doesn't like the undefined as any